mp/src/game/server/sdk/bots/bot_main.cpp

```diff
@@ -161,7 +161,11 @@ CBasePlayer *BotPutInServer( bool  bFrozen, const char *name )
 	ClientPutInServerOverride( &CBotManager::ClientPutInServerOverride_Bot );
 	if (!name)
 	{
-		name = g_aszBotNames[random->RandomInt(0, iNumBotNames - 1)];
+		for (int i = 0; i < 5; ++i) {
```

Comment:
Why not loop up to the max server size plus one?

Reply:
One name can be picked multiple times, so that's not as effective as you might think.
And 5 rounds are enough to get the probability tiny.